R/data-set.R

```diff
@@ -51,76 +60,166 @@ DataSet <- R6::R6Class(
+    },
+
+    #' @field yErrorType Type of the error - geometric or arithmetic.
+    #' When changing from arithmetic to geometric error, the values are considered in as fraction (1 = 100%).
+    #' When changing from geometric to arithmetic, the values are set to the same unit as \code{yErrorUnit}.
+    yErrorType = function(value) {
+      if (missing(value)) {
+        dataInfo <- rClr::clrGet(private$.yErrorValues$ref, "DataInfo")
+        errorTypeEnumVal <- rClr::clrGet(dataInfo, "AuxiliaryType")
+        return(netEnumName("OSPSuite.Core.Domain.Data.AuxiliaryType", errorTypeEnumVal))
+      }
+      private$.setErrorType(value)
+    },
+    #' @field yErrorUnit Unit in which the yErrorValues are defined. For arithmetic error, the unit must be valid
+    #' for \code{yDimension}. For geometric error, the unit must be valid for \code{Dimensionless}.
+    yErrorUnit = function(value) {
+      if (missing(value)) {
+        return(private$.yErrorValues$displayUnit)
+      }
+      private$.yErrorValues$displayUnit <- value
+    },
+    #' @field yErrorValues Values of error stored in the yErrorUnit unit
+    yErrorValues = function(values) {
+      if (missing(values)) {
+        return(private$.getColumnValues(private$.yErrorValues))
       }
 
-      private$setColumnValues(private$.yValues, values)
+      validateIsSameLength(self$yValues, values)
+      private$.setColumnValues(private$.yErrorValues, values)
+    },
+
+    #' @field metaData Returns a named list of meta data defined for the data set.
+    metaData = function(value) {
+      if (missing(value)) {
+        return(private$.dataRepository$metaData)
+      }
+      private$throwPropertyIsReadonly("metaData")
     }
   ),
   public = list(
-    dataRepository = NULL,
     #' @description
     #' Initialize a new instance of the class
-    #' @param dataRepository Instance of the \code{DataRepository} object to wrap
-    #' @return A new `DotNetWrapper` object.
+    #' @param dataRepository Instance of the \code{DataRepository} object to wrap.
+    #' If \code{NULL}, an empty \code{DataRepository} is created.
+    #' @return A new `DataSet` object.
     initialize = function(dataRepository = NULL) {
-      self$dataRepository <- dataRepository %||% private$createDataRepository()
-      private$initializeCache()
+      private$.dataRepository <- dataRepository %||% private$.createDataRepository()
+      private$.initializeCache()
     },
+
+    #' @description
+    #' Adds a new entry to meta data list or changes its value if the name is already present.
+    #' If \code{value} is \code{NULL}, the entry with corresponding name is deleted from meta data set.
+    #'
+    #' @param name Name of new meta data list entry
+    #' @param value Value of new meta data list entry
+    addMetaData = function(name, value) {
+      if (length(name) != 1) {
+        stop(messages$errorMultipleMetaDataEntries())
+      }
+      validateIsString(name)
+
+      # TODO

```

Comment:
Not implemented yet - requires the last Core .dll. And I am not sure how the deletion will work.